aspnetcore/blazor/file-uploads.md

```diff
@@ -45,7 +45,10 @@ To read data from a user-selected file, call <xref:Microsoft.AspNetCore.Componen
 
 <xref:Microsoft.AspNetCore.Components.Forms.IBrowserFile.OpenReadStream%2A> enforces a maximum size in bytes of its <xref:System.IO.Stream>. Reading one file or multiple files larger than 512,000 bytes (500 KB) results in an exception. This limit prevents developers from accidentally reading large files in to memory. The `maxAllowedSize` parameter of <xref:Microsoft.AspNetCore.Components.Forms.IBrowserFile.OpenReadStream%2A> can be used to specify a larger size if required.
 
-Avoid reading the incoming file stream directly into memory. For example, don't copy file bytes into a <xref:System.IO.MemoryStream> or read as a byte array. These approaches can result in performance and security problems, especially in Blazor Server. Instead, consider copying file bytes to an external store, such as a blob or a file on disk. If you need access to a <xref:System.IO.Stream> that represents the file's bytes, use <xref:Microsoft.AspNetCore.Components.Forms.IBrowserFile.OpenReadStream%2A?displayProperty=nameWithType>.
+If you need access to a <xref:System.IO.Stream> that represents the file's bytes, use <xref:Microsoft.AspNetCore.Components.Forms.IBrowserFile.OpenReadStream%2A?displayProperty=nameWithType>. Avoid reading the incoming file stream directly into memory all at once. For example, don't copy all of the file's bytes into a <xref:System.IO.MemoryStream> or read the entire stream into a byte array all at once. These approaches can result in performance and security problems, especially for Blazor Server apps. Instead, consider adopting either of the following approaches:
+
+* Copy file bytes to an external store, such as a blob or file on disk.
+* Upload files from the client directly to an external service. For an example, see the [Alternative approach with Azure Files](#alternative-approach-with-azure-files) section.
```

Comment:
These two options seem very similar - as both represent examples of uploading bytes to an external service, rather than to the server. Am I missing something?

Reply:
Good question. That came from the original text provided by the PU. I took it to mean that the file bytes are copied to an external store **_but still handled directly by the app_**. I perhaps have misinterpreted what this means. In the second case, I was hoping to call out that the app doesn't handle the bytes directly, unless later it fetches them from the external store on its own. Should I need to research the remark further so that we can see which PU member provided it? ... and then ask them what they meant.

Reply:
🔍 **_Found it_** 🔍 ... It was @pranavkm-supplied text.

TL;DR @pranavkm ... For the following remark in **BOLD** text in the Blazor File Uploads topic ...

> Avoid reading the incoming file stream directly into memory. For example, don't copy file bytes into a \<xref:System.IO.MemoryStream> or read as a byte array. These approaches can result in performance and security problems, especially in Blazor Server. **Instead, consider copying file bytes to an external store, such as a a blob or a file on disk.**

Cross-reference: https://github.com/dotnet/AspNetCore.Docs/pull/20321/files

Did you mean that the app handles the file bytes directly and sends them to an external store? Or, did you mean that the client uploads file bytes directly to an external store? It's the "copying" word that's obscuring it for me (i.e., "streaming" or "sending" feels like the client is doing it, while "copying" feels like the app is doing it).

We're trying to figure out if your remark is the same or different than my added bullet about perhaps using Azure Files so that clients can use a JS client/REST API to directly send file bytes to an external store without the app handling them at all.

Reply:
> Or, did you mean that the client uploads file bytes directly to an external store? It's the "copying" word that's obscuring it for me [i.e., "streaming" or "sending" feels like the client is doing it, while "copying" feels like the app (server-side) is doing it].

The latter. The idea is that you don't read arrays of bytes in to memory, because it increases GC pressure, could result in Large-Object-Heap allocations etc. Perhaps we could phrase the bullet points as:

* Upload files from the client directly to an external service. For an example, see the [Alternative approach with Azure Files](#alternative-approach-with-azure-files) section.
* Copy the stream directly to file on disk without reading it in to memory.

Reply:
Thx @pranavkm ... I'll use your exact bullet point language.